### PR TITLE
Burn wound surgery

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

reopening this https://github.com/ParadiseSS13/Paradise/pull/21801

Makes curing burn wounds its own separate surgery requiring an advanced trauma kit or ointment.

## Why It's Good For The Game
As discussed in discord, this brings burn wound surgery in line with the speed and ease of curing IB and broken bones. Debridement surgery is untouched and still cures burn wounds.

## Testing

compiled, did surgery

## Changelog
:cl:
add: Adds burn wounds surgery to fix critical burns, similar to internal bleeding surgery. Scalpel -> Hemostat -> Retractor -> Advanced burn kit -> Cautery.
/:cl:

### CODE REVIEW

**Comment by farie82** [comment]
Please add the original description or your own, so it is easier to see at a quick glance what this PR does.

File: code/modules/surgery/other.dm [hunk 1]
```diff
@@ -136,8 +208,8 @@
 /datum/surgery_step/fix_dead_tissue/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	var/obj/item/organ/external/affected = target.get_organ(target_zone)
 	user.visible_message(
-		"<span class='warning'> [user]'s hand slips, slicing an artery inside [target]'s [affected.name] with \the [tool]!</span>",
-		"<span class='warning'> Your hand slips, slicing an artery inside [target]'s [affected.name] with \the [tool]!</span>"
+		"<span class='warning'>[user]'s hand slips, slicing an artery inside [target]'s [affected.name] with \the [tool]!</span>",
+		"<span class='warning'>Your hand slips, slicing an artery inside [target]'s [affected.name] with \the [tool]!</span>"
```
Comment (lewcc) [inline code/modules/surgery/other.dm:212]:
```suggestion
		"<span class='warning'>[user]'s hand slips, slicing an artery inside [target]'s [affected.name] with [tool]!</span>",
		"<span class='warning'>Your hand slips, slicing an artery inside [target]'s [affected.name] with [tool]!</span>"
```

File: code/modules/surgery/other.dm [hunk 2]
```diff
@@ -94,13 +113,66 @@
...
+
+/datum/surgery_step/treat_burns/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	if(!(affected.status & ORGAN_BURNT))
+		to_chat(user, "<span class='warning'>The skin on [target]'s [parse_zone(affected)] seems to be in perfect condition, it doesn't need treatment.</span>")
+		return SURGERY_BEGINSTEP_SKIP
+
+	user.visible_message(
+		"[user] starts to treat affected tissue in [target]'s [affected.name] with [tool].",
+		"You start to treat affected tissue in [target]'s [affected.name] with [tool]."
```
Comment (lewcc) [inline code/modules/surgery/other.dm:142]:
```suggestion
		"[user] starts to treat the scorched tissue in [target]'s [affected.name] with [tool].",
		"You start to treat the scorched tissue in [target]'s [affected.name] with [tool]."
```

File: code/modules/surgery/abstract_steps.dm [hunk 3]
```diff
@@ -292,13 +292,31 @@
...
+
+/datum/surgery/intermediate/treat_burns/can_start(mob/user, mob/living/carbon/target)
+	. = ..()
+	if(!.)
+		return FALSE
+	var/mob/living/carbon/human/H = target
+	var/obj/item/organ/external/affected = H.get_organ(user.zone_selected)
+	if(affected.status & ORGAN_BURNT)
+		return TRUE
+	to_chat(user, "<span class='warning'>The skin on [target]'s [parse_zone(affected)] seems to be in perfect condition, it doesn't need treatment.</span>")
```
Comment (lewcc) [inline code/modules/surgery/abstract_steps.dm:309]: I'm pretty sure you don't need this here and can let the begin step handle the message. Sending messages in the `can_start` is a little bit of a no-no.
```suggestion
```

**Review by lewcc — CHANGES_REQUESTED** [review]

**Review by lewcc — CHANGES_REQUESTED** [review]
From what I can see, the surgery logic looks good. Just one note on the name of the surgery.

**Review by lewcc — APPROVED** [review]

**Pull request merged**